pkg/apiserver/rest/dualwriter_mode3.go

```diff
@@ -37,3 +38,24 @@ func (d *DualWriterMode3) Create(ctx context.Context, obj runtime.Object, create
 	}
 	return created, nil
 }
+
+// Get overrides the default behavior of the Storage and retrieves an object from Unified Storage
+// the object is still fetched from Legacy Storage if there is an error with Unified Storage
+func (d *DualWriterMode3) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	legacy, ok := d.legacy.(rest.Getter)
+	if !ok {
+		return nil, fmt.Errorf("legacy storage rest.Getter is missing")
+	}
+
+	s, err := d.Storage.Get(ctx, name, &metav1.GetOptions{})
+	if err == nil {
+		return s, err
+	}
+	if !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	fmt.Println("getting object from Unified Storage failed. Getting it from Legacy Storage.")
```

Comment:
We'll only hit this branch if it's not found, if it failed (errored) we'll return above.
```suggestion
	fmt.Println("Resource not found in Unified Storage. Getting it from Legacy Storage.")
```

Reply:
I'm actually not sure that we should have this fallback here, in theory at least every object should have been synced to US while in mode 2 so if we had to fall back that would indicate a serious error.